modules/shortcodes/vimeo.php

```diff
@@ -172,6 +172,77 @@ function vimeo_shortcode( $atts ) {
 
 add_shortcode( 'vimeo', 'vimeo_shortcode' );
 
+/**
+ * Filters the Vimeo shortcode to be AMP-compatible.
+ *
+ * @param string $html The shortcode HTML.
+ * @param string $shortcode_tag The shortcode's tag (name).
+ * @param array  $attr The attributes of the shortcode.
+ * @return string The filtered HTML.
+ */
+function amp_vimeo_shortcode( $html, $shortcode_tag, $attr ) {
+	if (
+		( class_exists( 'Jetpack_AMP_Support' ) && ! Jetpack_AMP_Support::is_amp_request() )
+		||
+		'vimeo' !== $shortcode_tag
+	) {
+		return $html;
+	}
+
+	$video_id = null;
+	if ( isset( $attr[0] ) ) {
+		$video_id = jetpack_shortcode_get_vimeo_id( $attr );
+	} elseif ( isset( $attr['id'] ) ) {
+		$video_id = $attr['id'];
+	}
+
+	list( $width, $height ) = jetpack_get_amp_vimeo_dimensions( $attr );
```

Comment:
Instead of doing a `list` followed by a `compact`, I wonder if it wouldn't be clearer / cheaper to create an array in the beginning and merge / push each value to it.

Reply:
This should be resolved through https://github.com/Automattic/jetpack/pull/13921/files/66e25acc900c5ec781765a06c15c640c0c61374b#diff-35236fb9fce8df90f9eea45af69cc92c, but let me know not.